### PR TITLE
Distance training with FTEs

### PR DESCRIPTION
**Pull request by sklum**
FTE templates shouldn't be used to train distances.

### CODE REVIEW

File: openbr/core/core.cpp [hunk 1]
```diff
@@ -86,6 +86,10 @@ struct AlgorithmCore
             qDebug("Projecting Enrollment");
             trainingWrapper->projectUpdate(data,data);
 
+            for (int i=data.size()-1; i>=0; i--)
+                if (data[i].file.fte || data[i].file.getBool("FTE"))
+                    data.removeAt(i);
```
Comment (caotto) [inline openbr/core/core.cpp]: Doing deletes of interior elements like this in a flat structure like a QList is typically not efficient (better to create a new list to contain non-deleted items, copy them out to the new list, drop the old list), although the perf impact may not be noticeable in this case.

Comment (jklontz) [inline openbr/core/core.cpp]: Do you have a source for that? A QList isn't a vector, so I would expect removeAt to be amortized O(1).

Comment (caotto) [inline openbr/core/core.cpp]: For example this article explicitly talks about QList

https://freininghaus.wordpress.com/2013/07/18/some-thoughts-on-algorithmic-complexity-part-2-be-careful-if-you-insert-into-or-remove-from-a-stdvector-qvector-or-qlist-in-a-loop/

How do you amortize deletes at arbitrary points, while maintaining integer
indexing?

Comment (jklontz) [inline openbr/core/core.cpp]: Thanks for the link

**Comment by sklum** [comment]
@caotto Better?

**Comment by caotto** [comment]
yes

**Pull request merged**